### PR TITLE
feat: add unused/deprecation diagnostic tags

### PR DESCRIPTION
**Pull request by digama0**
This makes the deprecation warning cause the text to be drawn in vscode with a strikethrough, and the unused variables linter warnings cause the text to be faded.

### CODE REVIEW

File: src/Init/Util.lean [hunk 1]
```diff
@@ -38,6 +38,7 @@ def dbgSleep {α : Type u} (ms : UInt32) (f : Unit → α) : α := f ()
 @[neverExtract, inline] def panicWithPosWithDecl {α : Type u} [Inhabited α] (modName : String) (declName : String) (line col : Nat) (msg : String) : α :=
   panic (mkPanicMessageWithDecl modName declName line col msg)
 
+set_option linter.unusedVariables.funArgs false in
```
Comment (Kha) [inline src/Init/Util.lean:41]: Mmh, why is this necessary now?

File: src/Lean/Widget/InteractiveDiagnostic.lean [hunk 2]
```diff
@@ -160,21 +152,19 @@ def msgToInteractiveDiagnostic (text : FileMap) (m : Message) (hasWidgets : Bool
     | none        => low
   let range : Range := ⟨low, high⟩
   let fullRange : Range := ⟨low, fullHigh⟩
-  let severity := match m.severity with
+  let severity? := match m.severity with
```
Comment (Kha) [inline src/Lean/Widget/InteractiveDiagnostic.lean:155]: Did the type change? If not, that's really confusing.

**Review by Kha — APPROVED** [review]
Would be great if you could put independent changes into separate commits or even better PRs.

File: src/Lean/Widget/InteractiveDiagnostic.lean (hunk 2 above)
Comment (digama0) [reply src/Lean/Widget/InteractiveDiagnostic.lean:155]: I just wanted it to have the same name as the field so I could use the abbreviated syntax for structure initialization at the end of the function

File: src/Init/Util.lean (hunk 1 above)
Comment (digama0) [reply src/Init/Util.lean:41]: I didn't look too deeply into it, but I was getting a warning here and there are similar `set_option` lines elsewhere in the file.

Comment (Kha) [reply src/Init/Util.lean:41]: I can't reproduce that. In fact, we now compile with `warningAsError`, so the CI should have failed on master if it happened while building Lean.

File: src/Lean/Widget/InteractiveDiagnostic.lean (hunk 2 above)
Comment (Kha) [reply src/Lean/Widget/InteractiveDiagnostic.lean:155]: Apparently it's always been `Option`. An explicit `some` sounds good!

File: src/Init/Util.lean (hunk 1 above)
Comment (digama0) [reply src/Init/Util.lean:41]: I'll remove it. There shouldn't be any other unrelated changes in this PR, except for the spacing fix and the comment typo. The import reductions may not be the minimum required but I could move those to a separate PR if desired.

Comment (leodemoura) [reply src/Init/Util.lean]: > I can't reproduce that. In fact, we now compile with `warningAsError`, so the CI should have failed on master if it happened while building Lean.

BTW, I am not sure whether the `warningAsError` is working on the CI. I have found "unused variable warnings" in the last few days, and I believe the CI did not report any problem. Could you please confirm whether this is working or not?
